core/src/main/java/com/datastrato/gravitino/meta/GroupEntity.java

```diff
@@ -24,17 +25,28 @@ public class GroupEntity implements Group, Entity, Auditable, HasIdentifier {
   public static final Field NAME =
       Field.required("name", String.class, "The name of the group entity.");
 
-  public static final Field ROLES =
-      Field.optional("roles", List.class, "The roles of the group entity.");
+  public static final Field ROLE_NAMES =
+      Field.optional("roleNames", List.class, "The role names of the group entity.");
+
+  public static final Field ROLE_IDS =
+      Field.optional("roleIds", List.class, "The role names of the group entity.");
 
   public static final Field AUDIT_INFO =
       Field.required("audit_info", AuditInfo.class, "The audit details of the group entity.");
 
+  public static final Field LATEST_GC_ROLES_TIME =
+      Field.required(
+          "latest_gc_roles_time",
+          Instant.class,
+          "The latest time when the entity removes invalid roles");
```

Comment:
Is it a global variable, or should it be per group/user?

Reply:
Every user/group entity will store its latest remove invalid roles time.